mlflow/deployments/base.py

```diff
@@ -126,72 +202,105 @@ def update_deployment(self, name, model_uri=None, flavor=None, config=None):
-    def list_deployments(self):
+    def list_deployments(self, endpoint_uri=None):
         """
         List deployments. This method is expected to return an unpaginated list of all
         deployments (an alternative would be to return a dict with a 'deployments' field
         containing the actual deployments, with plugins able to specify other fields, e.g.
         a next_page_token field, in the returned dictionary for pagination, and to accept
         a `pagination_args` argument to this method for passing pagination-related args).
 
+        :param endpoint_uri: URI of the Endpoint to list all contained deployments. May not be
+                             supported by all targets.
+
         :return: A list of dicts corresponding to deployments. Each dict is guaranteed to
                  contain a 'name' key containing the deployment name. The other fields of
                  the returned dictionary and their types may vary across deployment targets.
         """
         pass
 
     @abc.abstractmethod
-    def get_deployment(self, name):
+    def get_deployment(self, name, endpoint_uri=None):
         """
         Returns a dictionary describing the specified deployment, throwing a
         py:class:`mlflow.exception.MlflowException` if no deployment exists with the provided
-        ID.
+        name.
         The dict is guaranteed to contain an 'name' key containing the deployment name.
         The other fields of the returned dictionary and their types may vary across
         deployment targets.
 
         :param name: ID of deployment to fetch
+        :param endpoint_uri: URI of the Endpoint containing the deployment to get. May not be
+                             supported by all targets.
         """
         pass
 
     @abc.abstractmethod
-    def predict(self, deployment_name, df):
+    def predict(self, uri, df):
         """
-        Compute predictions on the pandas DataFrame ``df`` using the specified deployment.
+        Compute predictions on the pandas DataFrame ``df`` using the specified endpoint or deployment.
         Note that the input/output types of this method matches that of `mlflow pyfunc predict`
         (we accept a pandas DataFrame as input and return either a pandas DataFrame,
         pandas Series, or numpy array as output).
 
-        :param deployment_name: Name of deployment to predict against
+        :param uri: The URI of the endpoint or deployment to predict against. This can take one of
+                    three forms:
+
+                    - ``endpoints:/<endpoint name>`` for predicting against the default deployment
+                    in an endpoint
+                    - ``deployments:/<endpoint name>/<deployment name>`` for predicting against a
+                    specific deployment in an endpoint
+                    - ``deployments:/<deployment name>`` for predicting against a deployment for
+                    targets which don't support endpoints
```

Comment:
Since the support for endpoint (new) and deployment (existing) is loose, we have not touched whether and how to officially support the use of the scheme `"endpoints:/"` or `"deployments:/"` in the MLflow URI. For example, how could we tell `deployments:/<endpoint name>/<deployment name>` from `deployments:/<deployment name>`? To do so would require both deployment name and endpoint name not contain any "/".

To make the client support broadly applicable. It might be better to not use URI here at all but simply specify a generic name that can be interpreted properly by plugins. I apologize for contradicting myself on the design doc comments.